Add blog post: Reviewing 2019 in Docs (#18662)
kubernetes/website: content/en/blog/_posts/2020-01-17-Docs-Review-2019.md

---
layout: blog
title: "Reviewing 2019 in Docs"
date: 2020-01-17
slug: reviewing-2019-in-docs
---

**Author:** Zach Corleissen (Cloud Native Computing Foundation)

Hi, folks! I'm one of the co-chairs for the Kubernetes documentation special interest group (SIG Docs). This blog post is a review of SIG Docs in 2019. Our contributors did amazing work last year, and I want to highlight their successes. 

> Although I review 2019 in this post, my goal is to point forward to 2020. I observe some trends in SIG Docs--some good, others troubling. I want to raise visibility before those challenges increase in severity.

Reviewer: ```suggestion
Although I review 2019 in this post, my goal is to point forward to 2020. I observe some trends in SIG Docs–some good, others troubling. I want to raise visibility before those challenges increase in severity.
```

## The good

There was much to celebrate in SIG Docs in 2019.

> Kubernetes docs started the year with three localizations in progress. By the end of the year, we ended with ten localizations available, four of which (Chinese, French, Japanese, Korean) are reasonably complete. The Korean and French teams deserve special mentions for their contributions to git best practices across all localizations (Korean team) and help bootstrapping other localizations (French team). 

Reviewer 2: ```suggestion
Kubernetes docs started the year with three localizations in progress. By the end of the year, we ended with ten localizations available, four of which (Chinese, French, Japanese, Korean) are reasonably complete. The Korean and French teams deserve special mentions for their contributions to git best practices across all localizations (Korean team) and help to bootstrap other localizations (French team).
```

Author: The trailing space needs to go, but "bootstrapping" seems appropriate here.

> Despite significant transition over the year, SIG Docs [improved its review velocity](https://k8s.devstats.cncf.io/d/44/pr-time-to-approve-and-merge?orgId=1&var-period=w&var-repogroup_name=SIG%20Docs&var-apichange=All&var-size_name=All&var-kind_name=All), with a median review time from PR open to merge of slightly greater than 24 hours. 

Reviewer: ```suggestion
Despite significant transition over the year, SIG Docs [improved its review velocity](https://k8s.devstats.cncf.io/d/44/pr-time-to-approve-and-merge?orgId=1&var-period=w&var-repogroup_name=SIG%20Docs&var-apichange=All&var-size_name=All&var-kind_name=All), with a median review time from PR open to merge of just over than 24 hours. 
```

Author: @celestehorgan Thanks for the review. Would you please say more about this edit? Also, nit about the nit... 😉 

> merge of **just over than** 24 hours.

Reviewer: "slightly greater" just sounded awkward. (And yes, I caught your nit about the nit upon second read-through).

Take it or leave it, I'm ambivalent :)

Author: Done. 👍

> Despite significant transition over the course of the year, SIG Docs [improved its review velocity](https://k8s.devstats.cncf.io/d/44/pr-time-to-approve-and-merge?orgId=1&var-period=w&var-repogroup_name=SIG%20Docs&var-apichange=All&var-size_name=All&var-kind_name=All), with a median review time from PR open to merge of slightly greater than 24 hours. 

Reviewer 2: ```suggestion
Despite significant transition over the year, SIG Docs [improved its review velocity](https://k8s.devstats.cncf.io/d/44/pr-time-to-approve-and-merge?orgId=1&var-period=w&var-repogroup_name=SIG%20Docs&var-apichange=All&var-size_name=All&var-kind_name=All), with a median review time from PR open to merge of slightly greater than 24 hours. 
```

### But first, a disclaimer

{{< caution >}}

It is very hard to predict, especially the future.
-Niels Bohr

{{< /caution >}}

Some of my predictions are almost certainly wrong, either by scope or degree. While my estimates are often based in community discussions and awareness, any errors are mine alone. 

That said...

### Effects in 2020

Current levels of function aren't self-sustaining. Even with a strong playbook, the release cycle still requires expert support from at least one (and usually two) chairs during every cycle. Without fail, each release breaks processes in new and unexpected ways, and it requires familiarity and expertise to diagnose and resolve the cycle. As chairs continue to cycle&mdash;and to be clear, regular transitions are part of a healthy community&mdash;we accrue the risks associated with a pool that lacks sufficient professional depth and employer support.

Oddly enough, one of the challenges to staffing is that the docs appear good enough. Based on site analytics and survey responses, readers are pleased with the quality of the docs. When folks visit the site, they generally find what they need and behave like satisfied visitors.

> The danger is that this will change over time: slowly with occasional losses of function, annoying at first then increasingly critical. The more time passes without adequate staffing, the more difficult and costly fixes will become.

Reviewer 2: ```suggestion
The danger is that this will change over time: slowly with occasional losses of function, annoying at first, then increasingly critical. The more time passes without adequate staffing, the more difficult and costly fixes will become.
```

Author: I'm sufficiently ambivalent about the extra comma to not fight for it. 😅

I suspect this is true because the challenges we face now at decent levels of reader satisfaction are already difficult to fix. API reference generation is complex and brittle; the site's UI is outdated; and our most consistent requests are for more tutorials, advanced concepts, and code samples, all of which require ongoing, dedicated time to create.

**Release support remains strong.**

The release team continues a solid habit of leaving each successive team with better support than the previous release. This mostly takes the form of iterative improvements to the [docs release playbook], producing better documentation and reducing siloed knowledge held by legacy release leads.

**Staleness accelerates.**

Conceptual content becomes less accurate or relevant as features change or deprecate. Tutorial content degrades for the same reason.

> Content structure will also degrade: the categories of concepts, tasks, and tutorials are legacy categories that may not best fit the needs of current readers, let alone future ones.

Reviewer 2: ```suggestion
The content structure will also degrade: the categories of concepts, tasks, and tutorials are legacy categories that may not best fit the needs of current readers, let alone future ones.
```
